Make addition of Debugger Plugin conditional on FLAG

caisq opened the pull request:
The import also becomes conditional, which should get rid of
ImportErrors related to futures and grpcio when the user
hasn't explicitly specified the --debugger_data_server_grpc_port
flag.

wchargin commented on tensorboard/main.py:
```diff
@@ -279,9 +272,26 @@ def ConstructDebuggerPluginWithGrpcPort(context):
+        debugger_plugin.listen(FLAGS.debugger_data_server_grpc_port)
+        return debugger_plugin
+      except ImportError:
+        raise ImportError(
```
Can we print this and then reraise the original error (just `raise`), to prevent confusion in case the `ImportError` is _not_ caused by futures or grpcio?

Similarly, can we move everything except for the `import` statement out of the `try` block?
```py
try:
  from … import …
except:
  raise …
tf.logging.info(…)
…
```

caisq commented:
@wchargin. I addressed your comments. 

1. The message (including traceback) from the original `ImportError` is now preserved.
2. Reduced the size of the `try` block, to include only the single import statement.

wchargin commented on tensorboard/main.py:
```diff
@@ -279,9 +272,26 @@ def ConstructDebuggerPluginWithGrpcPort(context):
+        # pylint: disable=line-too-long,g-import-not-at-top
+        from tensorboard.plugins.debugger import debugger_plugin as debugger_plugin_lib
+        # pylint: enable=line-too-long,g-import-not-at-top
+      except ImportError as err:
```
Reraising like this destroys the traceback. Observe: the program
```python
import threading
import six
import sys


def a():
  return b()
def b():
  return c()
def c():
  raise ValueError('Oh, dear!')

def with_explicit_raise():
  try:
    a()
  except ValueError as e:
    raise ValueError(e.message + ' There was a problem.')


def with_six_reraise():
  try:
    a()
  except ValueError as e:
    (unused_type, unused_value, traceback) = sys.exc_info()
    six.reraise(
        ValueError,
        ValueError(e.message + ' There was a problem.'),
        traceback)


if __name__ == '__main__':
  for fn in (with_explicit_raise, with_six_reraise):
    t = threading.Thread(name=fn.__name__, target=fn)
    t.start()
```
yields
```
Exception in thread with_explicit_raise:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/tmp/z.py", line 17, in with_explicit_raise
    raise ValueError(e.message + ' There was a problem.')
ValueError: Oh, dear! There was a problem.

Exception in thread with_six_reraise:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/tmp/z.py", line 28, in with_six_reraise
    traceback)
  File "/tmp/z.py", line 22, in with_six_reraise
    a()
  File "/tmp/z.py", line 7, in a
    return b()
  File "/tmp/z.py", line 9, in b
    return c()
  File "/tmp/z.py", line 11, in c
    raise ValueError('Oh, dear!')
ValueError: Oh, dear! There was a problem.
```
Try using the latter formulation instead. (But with `ImportError`s instead of `ValueError`s.)

caisq replied:
Thanks. Done.
